src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/models/ExtractRequest.java

```diff
@@ -7,39 +7,64 @@
 
     private Extract extract;
     private Document document;
+    private Long receivedTimeMillis;
+    private Long processedTimeMillis;
 
     public ExtractRequest(
             @JsonProperty("extract") Extract extract,
-            @JsonProperty("document") Document document) {
+            @JsonProperty("document") Document document,
+            @JsonProperty("receivedTimeMillis") Long receivedTimeMillis,
+            @JsonProperty("processedTimeMillis") Long processedTimeMillis) {
         this.extract = extract;
         this.document = document;
+        this.receivedTimeMillis = receivedTimeMillis;
+        this.processedTimeMillis = processedTimeMillis;
     }
 
     public ExtractRequest(Builder builder) {
         this.document = builder.document;
         this.extract = builder.extract;
     }
 
+    public void updateProcessedTimeMillis() {
+        this.processedTimeMillis = System.currentTimeMillis();
+    }
+
+    public long getProcessingTimeMillis() {
+
+        if(processedTimeMillis == null || receivedTimeMillis == null) {
+            return -1l;
```

Comment:
magic number?

Reply:
That is long syntax. Add an L to the end.